WordPress/src/main/java/org/wordpress/android/ui/mysite/SelectedSiteSource.kt

```diff
@@ -9,12 +15,40 @@ import javax.inject.Singleton
 
 @Singleton
 class SelectedSiteSource @Inject constructor(
-    private val selectedSiteRepository: SelectedSiteRepository
+    private val selectedSiteRepository: SelectedSiteRepository,
+    private val dispatcher: Dispatcher
 ) : MySiteSource<SelectedSite> {
+    val refresh: MutableLiveData<Boolean> = MutableLiveData<Boolean>(false)
+
     override fun buildSource(
         coroutineScope: CoroutineScope,
         siteLocalId: Int
     ) = selectedSiteRepository.selectedSiteChange
             .filter { it == null || it.id == siteLocalId }
             .map { SelectedSite(it) }
+
+    fun refresh() {
+        selectedSiteRepository.updateSiteSettingsIfNecessary()
+        selectedSiteRepository.getSelectedSite()?.let {
+            refresh.postValue(true)
+            dispatcher.dispatch(SiteActionBuilder.newFetchSiteAction(it))
+        }
+    }
+
+    fun isRefreshing() = refresh.value as Boolean
```

Comment:
Suggestion (💡): Consider adding some additional test to verify `refresh` state changes as expected. For example:
- When on `buildSource(...)`, it should be `false`
- When on `refresh()`, with a `selected site`, it should be `true`
- When on `refresh()`, without a `selected site`, it should be `false`
- When on `onSiteChanged(...)`, it should be `false` (from `true`)

Reply:
Good idea. I hadn't planned on adding those tests until implementation of the refresh itself, but it doesn't hurt to do them now and then update if needed. Added new tests in 76fe64a

Reply:
👋 @zwarm !

This is awesome, thanks so much for adding those extra test! 🥇